Source/engine.h

```diff
@@ -37,35 +37,67 @@
 #include "engine/size.hpp"
 #include "engine/surface.hpp"
 #include "miniwin/miniwin.h"
+#include "utils/stdcompat/apply.hpp"
 #include "utils/stdcompat/cstddef.hpp"
 
 #define TILE_WIDTH 64
 #define TILE_HEIGHT 32
 
 namespace devilution {
 
-template <typename V, typename X>
-bool IsAnyOf(const V &v, X x)
-{
-	return v == x;
-}
+namespace {
 
-template <typename V, typename X, typename... Xs>
-bool IsAnyOf(const V &v, X x, Xs... xs)
-{
-	return IsAnyOf(v, x) || IsAnyOf(v, xs...);
-}
+template <typename... Args>
+struct AnyOfHelper {
+	std::tuple<Args...> values;
 
-template <typename V, typename X>
-bool IsNoneOf(const V &v, X x)
-{
-	return v != x;
-}
+	constexpr AnyOfHelper(Args... values)
+	    : values(std::move(values)...)
+	{
+	}
 
-template <typename V, typename X, typename... Xs>
-bool IsNoneOf(const V &v, X x, Xs... xs)
+	template <typename T>
+	[[nodiscard]] friend constexpr bool operator==(T lhs, AnyOfHelper const &rhs) noexcept
+	{
+		return apply(
+		    [&](auto... vals) { return ((lhs == vals) || ...); },
+		    rhs.values);
+	}
+
+	template <typename T>
+	[[nodiscard]] friend constexpr bool operator!=(T lhs, AnyOfHelper const &rhs) noexcept
+	{
+		return apply(
+		    [&](auto... vals) { return ((lhs != vals) && ...); },
+		    rhs.values);

```

Comment:
This might as well be an inversion of operator==
```suggestion
		return !(lhs == rhs);
```